flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/util/VertexParallelismAndInputInfosDeciderUtils.java

```diff
@@ -658,4 +660,148 @@ public static long calculateDataVolumePerTaskForInput(
             long globalDataVolumePerTask, long inputsGroupBytes, long totalDataBytes) {
         return (long) ((double) inputsGroupBytes / totalDataBytes * globalDataVolumePerTask);
     }
+
+    public static Optional<String> constructOptimizationLog(
```

Comment:
Could we print the optimization log in this method and split this long method into several smaller methods?